controllers/topology/blueprint.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/scope"
+	"sigs.k8s.io/cluster-api/util/patch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// getBlueprint gets a ClusterBlueprint with the ClusterClass and the referenced templates to be used for a managed Cluster topology.
+// It also converts and patches all ObjectReferences in ClusterClass and ControlPlane to the latest apiVersion of the current contract.
+// NOTE: This function assumes that cluster.Spec.Topology.Class is set.
+func (r *ClusterReconciler) getBlueprint(ctx context.Context, s *scope.Scope) (_ *scope.ClusterBlueprint, reterr error) {
+	cluster := s.Current.Cluster
+	blueprint := &scope.ClusterBlueprint{
+		Topology:           cluster.Spec.Topology,
+		ClusterClass:       &clusterv1.ClusterClass{},
+		MachineDeployments: map[string]*scope.MachineDeploymentBlueprint{},
```

Comment:
```suggestion
                 ControlPlane : &scope.ControlPlaneBlueprint{},
		MachineDeployments: map[string]*scope.MachineDeploymentBlueprint{},
```
also initialize ControlPlane?

Reply:
This is currently done down in the code; we can eventually reshuffle things in a follow up